src/drawing/DrawingPathGroup.tsx

```diff
@@ -13,31 +13,45 @@ export type Props = {
 
 export const DRAWING_BORDER_WIDTH = 1;
 
-export const DrawingPathGroup = ({
-    isActive = false,
-    pathGroup: {
-        paths,
-        stroke: { color, size },
-    },
-    rootEl,
-}: Props): JSX.Element => {
-    let strokeWidth = size;
+export const getStrokeWidths = (
+    initialWidth: number,
+    referenceEl: DrawingSVGRef | null,
+): { strokeWidth: number; strokeWidthWithBorder: number } | null => {
+    let strokeWidth = initialWidth;
     // A line has two sides and each side needs a border, so add DRAWING_BORDER_WIDTH * 2
     let strokeWidthWithBorder = strokeWidth + DRAWING_BORDER_WIDTH * 2;
 
     // "vector-effect: non-scaling-stroke" prevents stroke width to be scaled
     // However, IE/Edge doesn't support it, so we have to manually inverse scale the stroke width
     if (!isVectorEffectSupported()) {
-        if (!rootEl) {
-            return <g className="ba-DrawingPathGroup" />;
+        if (!referenceEl) {
+            return null;
         }
-        const { clientWidth } = rootEl;
+        const { clientWidth } = referenceEl;
         const scale = clientWidth / 100; // SVG viewbox is 100 * 100
         strokeWidth /= scale;
         strokeWidthWithBorder /= scale;
     }
 
-    return (
+    return {
+        strokeWidth,
+        strokeWidthWithBorder,
+    };
+};
+
+export const DrawingPathGroup = ({
+    isActive = false,
+    pathGroup: {
+        paths,
+        stroke: { color, size },
+    },
+    rootEl,
+}: Props): JSX.Element => {
+    const { strokeWidth, strokeWidthWithBorder } = getStrokeWidths(size, rootEl) ?? {};
+
+    return !strokeWidth || !strokeWidthWithBorder ? (
+        <g className="ba-DrawingPathGroup" />
```

Comment:
Can we return the main `<g>` element and do this check before `paths.map` to avoid duplicating the element? If not, can we make this a guard clause to avoid a ternary return?